digits/model/forms.py

```diff
@@ -339,6 +339,23 @@ def validate_custom_network_snapshot(form, field):
         tooltip="The job won't start until all of the chosen GPUs are available."
     )
 
+    # Select 1 of several GPUs
+    select_one_of_gpus = utils.forms.SelectField(
+        'Select which GPU you would like to use',
+        choices=[('next', 'Next available')] + [(
+            index,
+            '#%s - %s (%s memory)' % (
+                index,
+                get_device(index).name,
+                sizeof_fmt(
+                    get_nvml_info(index)['memory']['total']
+                    if get_nvml_info(index) and 'memory' in get_nvml_info(index)
+                    else get_device(index).totalGlobalMem)
+            ),
+        ) for index in config_value('gpu_list').split(',') if index],
+        default='next',
```

Comment:
@lukeyeager the same behavior that you'd get if you were to start a training job without a GPU:
1. If all setup was made with CPU only, you'd get the same page you currently get (without the multi-gpu form)
2. If you manage to have Caffe compiled with CUDA and has no GPU on the system (suppose you removed it afterwards -- I tested it masking all my GPUs using an invalid id on `CUDA_VISIBLE_DEVICES`), then you'd get the page without multi-gpu form and your job will fail exactly as the train job fails currently:

```
Check failed: error == cudaSuccess (38 vs. 0)  no CUDA-capable device is detected
```